### ISSUE TITLE
Update to latest SDK Reference

### ISSUE THREAD

**Issue by grokify** [ringcentral/ringcentral-csharp#19]
Update interfaces to latest SDK reference:

- https://github.com/ringcentral/sdk-reference

The current version is based on an earlier version of the SDK reference. Some changes include:

* Update of read accessor names to remove preceding `Get` or `get`. For example, `GetJson()` should now be `Json()`
* Addition of [`ApiResponse`](https://github.com/ringcentral/sdk-reference/blob/master/http/ApiResponse.ts) to provide a generic return object.

**Comment by tylerlong** [comment]
I have been working on this in a forked repository:  https://github.com/tylerlong/ringcentral-csharp

Hopefully I can create a PR and merge my work into this repository in the near future.

**Comment by grokify** [comment]
PR #22 has been merged.

Some things I noted include:

| Existing | New |
|------------|-------|
| `Platform.Authorize()` | `Platform.Login()` |
| `ApiResponse.GetBody()` | `ApiResponse.Body()` |
| `ApiResponse.GetJson()` | `ApiResponse.Json()` |
| `ApiResponse.GetStatus()` | `ApiResponse.Status()` |
| `ApiResponse.GetError()` | `ApiResponse.Error()` |

**Comment by tylerlong** [comment]
I will take a look as soon as I finish the task on hand. I have an concern regarding the renaming. Can we make them properties instead of methods?

For example:

Existing | New Option 1 | New Option 2
  ---|---|---
`ApiResponse.GetBody()` | `ApiResponse.Body()` | `ApiResponse.Body`

Acutally I already did similar thing to `SDK.Platform`. It's not `SDK.Platform()`. Because I think it should be a property instead of a method.

Ref: https://msdn.microsoft.com/en-us/library/ms229054(v=vs.100).aspx

> Do use a property, rather than a method, if the value of the property is stored in the process memory and the property would just provide access to the value.

**Comment by grokify** [comment]
I noticed that `SDK.Platform` was changed. I'm okay with the change where it makes sense. `ApiResponse` methods above seem to make sense as can `SubscriptionServiceImplementation.Events()`.

To make `SubscriptionServiceImplementation.FullEventsFilter()` a property, it seems like it would need to be updated every time `SubscriptionServiceImplementation.Events` was updated so take that into consideration.
